pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -786,28 +786,65 @@ func (desc *Immutable) HydrateTypeInfoWithName(
 	}
 }
 
+// NumEnumMembers returns the number of enum members if the type is an
+// enumeration type, 0 otherwise.
+func (desc *immutable) NumEnumMembers() int {
+	return len(desc.EnumMembers)
+}
+
+// GetMemberPhysicalRepresentation returns the physical representation of the
+// enum member at ordinal enumMemberOrdinal.
+func (desc *immutable) GetMemberPhysicalRepresentation(enumMemberOrdinal int) []byte {
+	return desc.physicalReps[enumMemberOrdinal]
+}
+
+// GetMemberLogicalRepresentation returns the logical representation of the enum
+// member at ordinal enumMemberOrdinal.
+func (desc *immutable) GetMemberLogicalRepresentation(enumMemberOrdinal int) string {
+	return desc.logicalReps[enumMemberOrdinal]
+}
+
+// IsMemberReadOnly returns true iff the enum member at ordinal
+// enumMemberOrdinal is read-only.
+func (desc *immutable) IsMemberReadOnly(enumMemberOrdinal int) bool {
+	return desc.readOnlyMembers[enumMemberOrdinal]
+}
+
+// NumReferencingDescriptors returns the number of descriptors referencing this
+// type, directly or indirectly.
+func (desc *immutable) NumReferencingDescriptors() int {
+	return len(desc.ReferencingDescriptorIDs)
+}
+
+// GetReferencingDescriptorID returns the ID of the referencing descriptor at
+// ordinal refOrdinal.
+func (desc *immutable) GetReferencingDescriptorID(refOrdinal int) descpb.ID {
+	return desc.ReferencingDescriptorIDs[refOrdinal]
+}
+
```

Comment:
These methods needed to be added to `catalog.TypeDescriptor` but they're straightforward getters.